syntax = "proto3";

package chip.rpc;

import 'proto/empty.proto';

// type lengths defined in chip spec 8.2.3.1
message DeviceInfo {
  uint32 vendor_id = 1;
  uint32 product_id = 2;
  uint32 software_version = 3;
  optional string serial_number = 4;
}

service Device {
  rpc FactoryReset(Empty) returns (Empty){}
  rpc Reboot(Empty) returns (Empty){}
  rpc TriggerOta(Empty) returns (Empty){}
  rpc GetDeviceInfo(Empty) returns (DeviceInfo){}
}
